cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/profile/ProfileStore.java

```diff
@@ -0,0 +1,325 @@
+    }
+    Profile newProfile = new Profile(oldProfile.getName(), oldProfile.getLabel(), oldProfile.getDescription(),
+                                     oldProfile.getScope(), expectedStatus, oldProfile.getProvisioner(),
+                                     oldProfile.getCreatedTsSeconds());
+    fields.add(Fields.stringField(StoreDefinition.ProfileStore.PROFILE_DATA_FIELD, GSON.toJson(newProfile)));
+    profileTable.upsert(fields);
+  }
+
+  /**
+   * Get assignments with the profile.
+   *
+   * @param profileId the profile id
+   * @return the set of entities that the profile is assigned to
+   * @throws NotFoundException if the profile is not found
+   */
+  public Set<EntityId> getProfileAssignments(ProfileId profileId) throws NotFoundException, IOException {
+    Collection<Field<?>> fields = getProfileKeys(profileId);
+    Profile profile = getProfileInternal(fields);
+    if (profile == null) {
+      throw new NotFoundException(profileId);
+    }
+
+    Set<EntityId> entities = new HashSet<>();
+    scanEntities(profileId, entities);
+    return Collections.unmodifiableSet(entities);
+  }
+
+  /**
+   * Add an assignment to the profile. Assignment can only be added if the profile is ENABLED
+   *
+   * @param profileId the profile id
+   * @param entityId the entity to add to the assgiment
+   */
+  public void addProfileAssignment(ProfileId profileId,
+                                   EntityId entityId) throws ProfileConflictException, NotFoundException, IOException {
+    Collection<Field<?>> fields = getProfileKeys(profileId);
+    Profile profile = getProfileInternal(fields);
+    if (profile == null) {
+      throw new NotFoundException(profileId);
+    }
+    if (profile.getStatus() == ProfileStatus.DISABLED) {
+      throw new ProfileConflictException(
+        String.format("Profile %s is DISABLED. No entity can be assigned to it.", profileId.getProfile()), profileId);
+    }
+    addEntityIdKey(fields, entityId);
+    fields.add(Fields.stringField(StoreDefinition.ProfileStore.ENTITY_DATA_FIELD, GSON.toJson(entityId)));
+    profileEntityTable.upsert(fields);
+  }
+
+  /**
+   * Remove an assignment from the profile.
+   *
+   * @param profileId the profile id
+   * @param entityId the entity to remove from the assignment
+   * @throws NotFoundException if the profile is not found
+   */
+  public void removeProfileAssignment(ProfileId profileId, EntityId entityId) throws NotFoundException, IOException {
+    Collection<Field<?>> keys = getProfileKeys(profileId);
+    Profile profile = getProfileInternal(keys);
+    if (profile == null) {
+      throw new NotFoundException(profileId);
+    }
+    addEntityIdKey(keys, entityId);
+    profileEntityTable.delete(keys);
+  }
+
+  @Nullable
+  private Profile getProfileInternal(Collection<Field<?>> profileKeys) throws IOException {
+    Optional<StructuredRow> rowOptional = profileTable.read(profileKeys);
+    //noinspection OptionalIsPresent
```

Comment:
remove comment? not sure what this means

Reply:
This is for intellij. Otherwise it shows a warning saying that the condition below can be converted to a functional style expression. I find the functional style expression harder to read in this case.

I changed the comment based suppression to an annotation based one